apps/challenges/aws_utils.py

```diff
@@ -697,3 +699,121 @@ def restart_workers_signal_callback(sender, instance, field_name, **kwargs):
                 instance.pk, field_name
             )
         )
+
+
+def create_eks_nodegroup(instance, cluster_name):
+    """
+    Creates a nodegroup when a EKS cluster is created by the EvalAI admin
+    """
+    nodegroup_name = "{}-nodegroup".format(instance.title.replace(" ", "-"))
+    client = get_boto3_client("eks", aws_keys)
+    # TODO: Move the hardcoded cluster configuration such as the instance_type, subnets, AMI to challenge configuration later.
+    try:
+        response = client.create_nodegroup(
+            clusterName=cluster_name,
+            nodegroupName=nodegroup_name,
+            scalingConfig={"minSize": 1, "maxSize": 100, "desiredSize": 1},
+            diskSize=20,
+            subnets=[VPC_DICT["SUBNET_1"], VPC_DICT["SUBNET_2"], ],
+            instanceTypes=["g4dn.xlarge", ],
+            amiType="AL2_x86_64_GPU",
+            nodeRole=settings.EKS_NODEGROUP_ROLE_ARN,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return response
+    waiter = client.get_waiter("nodegroup_active")
+    waiter.wait(name=nodegroup_name)
+    # create_eks_deployment()
+
+
+def create_eks_cluster(sender, instance, field_name, **kwargs):
+    """
+    Called when Challenge is approved by the EvalAI admin
+    """
+    from .models import ChallengeEvaluationCluster
```

Comment:
Why are we importing it in the function?

Reply:
![cyclic](https://user-images.githubusercontent.com/19167324/83355146-27aa5280-a37b-11ea-8bda-9ca20842659f.png)
When importing it outside the function, it was a cyclic import which gave the above error. I have removed this for now, will add it in the next PR

Reply:
Can you add in this PR?
Its ok to have imports here because cyclic imports
From [Python Docs](https://docs.python.org/3.8/tutorial/modules.html#more-on-modules)
>It is customary but not required to place all import statements at the beginning of a module (or script, for that matter). The imported module names are placed in the importing module’s global symbol table.